[
    {
        "type": "message",
        "user": "U1JAPLQSY",
        "text": "BTW, for hyperledger... if i want to ask a unanswered question, should we ask in StackOverflow instead of here?",
        "ts": "1468566742.002295"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "you can try here and try SO, up to you :slightly_smiling_face:",
        "ts": "1468566809.002296"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U1LS81ER3>: sorry, not having much luck deploying chaincode that isn't in the hyperledger folder :confused:",
        "ts": "1468567233.002297"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "it looks like my `peer chaincode deploy` works but then I get this in a block where I expect the deploy to happen\n&lt;error&gt;Failed to launch chaincode spec(Error starting container: The command '\/bin\/sh -c go install <http:\/\/github.com\/IBM-Blockchain\/learn-chaincode\/blob\/master\/finished|github.com\/IBM-Blockchain\/learn-chaincode\/blob\/master\/finished> &amp;amp;&amp;amp; cp src\/github.com\/hyperledger\/fabric\/peer\/core.yaml $GOPATH\/bin &amp;amp;&amp;amp; mv $GOPATH\/bin\/finished $GOPATH\/bin\/842b56c1a37dbe53282da0885c126ad4d2e52b885a1ac577e7abd1cb6b8e4bb15ddced4db7d5c9f05f1afbb7d993448cc3ee9cf5d46eadb87f1a0f5d899254ed' returned a non-zero code: 2)&lt;\/error&gt;",
        "ts": "1468567286.002298"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "<@U1LS81ER3>: so I seem to get a transactions even when I have the wrong function name... maybe you're on a slightly different commit level of hyperledger to me though\n\nbad function: \n{\"transactions\":[{\"type\":2,\"chaincodeID\":\"Eo...jA=\",\"payload\":\"Cp...Zw==\",\"uuid\":\"d1cd8f76-61ea-4c28-8e84-217c48c6c901\",\"timestamp\":{\"seconds\":1468569313,\"nanos\":770787480}}],\"stateHash\":\"iDI+Jxun66wZT7s1BzXo\/Qt6WHAv+lIz7CGVZ1GhCCSee0mVBF1+3dcw8S7S3rFpTSi5KwXRr6OcOLTT+HoP1w==\",\"previousBlockHash\":\"7e\/e1D9brUB0\/DHtlwHvepki\/zHzAnC7RPR1RdC+PlTOeUepj5aqPT8fBC+zV\/jK8X6dXe+C8VCZo6caqmpt+g==\",\"consensusMetadata\":\"CA8=\",\"nonHashData\":{\"localLedgerCommitTimestamp\":{\"seconds\":1468569313,\"nanos\":782345226},\"transactionResults\":[{\"uuid\":\"d1cd8f76-61ea-4c28-8e84-217c48c6c901\",\"errorCode\":1,\"error\":\"Transaction or query returned with failure: Received unknown function invocation\"}]}}\n\ngood function:\n{\"transactions\":[{\"type\":2,\"chaincodeID\":\"Eo...jA=\",\"payload\":\"Cp...5n\",\"uuid\":\"936febc9-3234-4224-9d80-a4ea52d58396\",\"timestamp\":{\"seconds\":1468569350,\"nanos\":48766998}}],\"stateHash\":\"iDI+Jxun66wZT7s1BzXo\/Qt6WHAv+lIz7CGVZ1GhCCSee0mVBF1+3dcw8S7S3rFpTSi5KwXRr6OcOLTT+HoP1w==\",\"previousBlockHash\":\"+M+98bpw6CnIXAl9hPw8qEhyE9\/xkNoHS8mdctC906ADDiWL7MLqEfHwETpMQpHsijSQiAAnp8GQxzO4VUXBlQ==\",\"consensusMetadata\":\"CBA=\",\"nonHashData\":{\"localLedgerCommitTimestamp\":{\"seconds\":1468569350,\"nanos\":73584796},\"transactionResults\":[{\"uuid\":\"936febc9-3234-4224-9d80-a4ea52d58396\"}]}}",
        "ts": "1468569567.002299"
    },
    {
        "type": "message",
        "user": "U1LES78TT",
        "text": "that was an `invoke` via the REST API if that makes any difference",
        "ts": "1468569629.002300"
    },
    {
        "type": "message",
        "user": "U1422GVM1",
        "text": "Hi, I'm trying to run 2 validating peers, which I could run successfully for the first time. But when I stopped them and restarted (CA service too), it threw an error as \"Identity or token does not match\". Below is the description of error shown in console.\nvagrant@hyperledger-devenv:v0.0.10-0581047:\/opt\/gopath\/src\/github.com\/hyperledger\/fabric$ docker run --rm -it -e CORE_VM_ENDPOINT=<http:\/\/172.17.0.1:2375> -e CORE_PEER_ID=test_v\np0 -e CORE_PEER_ADDRESSAUTODETECT=true -e CORE_SECURITY_ENABLED=true -e CORE_SECURITY_PRIVACY=true -e CORE_PEER_PKI_ECA_PADDR=172.17.0.1:50051 -e CORE_PEER_PKI_TCA_PADDR=172.1\n7.0.1:50051 -e CORE_PEER_PKI_TLSCA_PADDR=172.17.0.1:50051 -e CORE_SECURITY_ENROLLID=test_vp0 -e CORE_SECURITY_ENROLLSECRET=MwYpmSRjupbT hyperledger\/fabric-peer peer node start\n2016\/07\/15 08:28:32 Load docker HostConfig: &amp;{Binds:[] CapAdd:[] CapDrop:[] ContainerIDFile: LxcConf:[] Privileged:false PortBindings:map[] Links:[] PublishAllPorts:false DNS:[] DNSSearch:[] ExtraHosts:[] VolumesFrom:[] NetworkMode:host IpcMode: PidMode: UTSMode: RestartPolicy:{Name: MaximumRetryCou\nnt:0} Devices:[] LogConfig:{Type: Config:map[]} ReadonlyRootfs:false SecurityOpt:[] CgroupParent: Memory:0 MemorySwap:0 MemorySwappiness:0 OOMKillDisable:false CPUShares:0 CPUSet: CPUSetCPUs: CPUSetMEMs: CPUQuota:0 CPUPeriod:0 BlkioWeight:0 Ulimits:[]}\n08:28:32.928 [peer] func1 -&gt; INFO 002 Auto detected peer address: 172.17.0.2:30303\n08:28:32.929 [peer] func1 -&gt; INFO 003 Auto detected peer address: 172.17.0.2:30303\n08:28:32.931 [main] serve -&gt; INFO 004 Security enabled status: true\n08:28:32.931 [eventhub_producer] start -&gt; INFO 006 event processor started\n08:28:32.931 [main] serve -&gt; INFO 005 Privacy enabled status: true\n08:28:32.931 [crypto] RegisterValidator -&gt; INFO 007 Registering validator [test_vp0] with name [test_vp0]...\n08:28:32.969 [crypto] Errorf -&gt; ERRO 008 [validator.test_vp0] Failed invoking CreateCertficatePair [rpc error: code = 2 desc = \"Identity or token does not match.\"].\n08:28:32.969 [crypto] Errorf -&gt; ERRO 009 [validator.test_vp0] Failed getting enrollment certificate [id=test_vp0]: [rpc error: code = 2 desc = \"Identity or token does not match.\"]\n08:28:32.969 [crypto] Errorf -&gt; ERRO 00a [validator.test_vp0] Failed retrieving enrollment data [rpc error: code = 2 desc = \"Identity or token does not match.\"].\n08:28:32.969 [crypto] Errorf -&gt; ERRO 00b [validator.test_vp0] Failed registering node crypto engine [rpc error: code = 2 desc = \"Identity or token does not match.\"].\n08:28:32.969 [crypto] Errorf -&gt; ERRO 00c [validator.test_vp0] Failed registering [test_vp0]: [rpc error: code = 2 desc = \"Identity or token does not match.\"]\n08:28:32.969 [crypto] Errorf -&gt; ERRO 00d [validator.test_vp0] Failed registering [test_vp0]: [rpc error: code = 2 desc = \"Identity or token does not match.\"]\n08:28:32.969 [crypto] RegisterValidator -&gt; ERRO 00e Failed registering validator [test_vp0] with name [test_vp0] [rpc error: code = 2 desc = \"Identity or token does not match.\"].\n\nUsage:\n  peer node start [flags]\n\nFlags:\n  -h, --help[=false]: help for start\n      --peer-chaincodedev[=false]: Whether peer in chaincode development mode\n\n\nGlobal Flags:\n      --logging-level=\"\": Default logging level and overrides, see core.yaml for full syntax\n      --test.coverprofile=\"coverage.cov\": Done\n\nError: rpc error: code = 2 desc = \"Identity or token does not match.\"\n08:28:32.969 [main] main -&gt; INFO 00f Exiting.....",
        "ts": "1468571807.002301"
    },
    {
        "type": "message",
        "user": "U1JAPLQSY",
        "text": "omg..",
        "ts": "1468573976.002302"
    },
    {
        "type": "message",
        "user": "U1RKVNEGH",
        "text": "the second time you run the validator, you need to delete \/var\/lib\/hyperledger\/production",
        "ts": "1468577216.002303"
    },
    {
        "type": "message",
        "user": "U1RKVNEGH",
        "text": "<@U1422GVM1>:",
        "ts": "1468577361.002304"
    },
    {
        "type": "message",
        "user": "U1RKVNEGH",
        "text": "I have the same issue, need to find a way out instead of remove the data",
        "ts": "1468577382.002305"
    },
    {
        "user": "U1S1RK0CB",
        "text": "<@U1S1RK0CB|sonke> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468581765.002306"
    },
    {
        "type": "message",
        "user": "U1CS08EP3",
        "text": "Hi , any one knows the root password of vagrant image which we are using in the Hyperledger ?",
        "ts": "1468581783.002307"
    },
    {
        "type": "message",
        "user": "U1CS08EP3",
        "text": "i meant the ubuntu image ..",
        "ts": "1468581807.002308"
    },
    {
        "type": "message",
        "user": "U1CS08EP3",
        "text": "i tried to access the docker images in \/var\/lib\/docker",
        "ts": "1468581872.002309"
    },
    {
        "type": "message",
        "user": "U1CS08EP3",
        "text": "so that i can move them to another machine which do not hv network connectivity..",
        "ts": "1468581894.002310"
    },
    {
        "type": "message",
        "user": "U1CS08EP3",
        "text": "but it s throwing a \"permission denied:",
        "ts": "1468581908.002311"
    },
    {
        "type": "message",
        "user": "U1K72JJHW",
        "text": "They are publicly available",
        "ts": "1468581921.002312"
    },
    {
        "type": "message",
        "user": "U1K72JJHW",
        "text": "<https:\/\/hub.docker.com\/r\/yeasy\/hyperledger-peer\/>",
        "ts": "1468581964.002313"
    },
    {
        "type": "message",
        "user": "U1K72JJHW",
        "text": "sudo works on the vagrant image",
        "ts": "1468582666.002314"
    },
    {
        "type": "message",
        "user": "U12452RAP",
        "text": "guys who have issue with CA and peers, this PR should helpful",
        "ts": "1468586976.002315",
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U0KM61BCP"
                ],
                "count": 1
            }
        ]
    },
    {
        "type": "message",
        "user": "U12452RAP",
        "text": "<https:\/\/github.com\/hyperledger\/fabric\/pull\/2150>",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "refactor documentation  by christo4ferris \u00b7 Pull Request #2150 \u00b7 hyperledger\/fabric \u00b7 GitHub",
                "title_link": "https:\/\/github.com\/hyperledger\/fabric\/pull\/2150",
                "text": "Description refactor of docs to make the use of fabric components more accessible to mere mortals and without requiring that binaries and images be built by the user. This turned into a major refa...",
                "fallback": "GitHub: refactor documentation  by christo4ferris \u00b7 Pull Request #2150 \u00b7 hyperledger\/fabric",
                "thumb_url": "https:\/\/avatars1.githubusercontent.com\/u\/3630697?v=3&s=400",
                "from_url": "https:\/\/github.com\/hyperledger\/fabric\/pull\/2150",
                "thumb_width": 400,
                "thumb_height": 400,
                "service_icon": "https:\/\/a.slack-edge.com\/e8ef6\/img\/unfurl_icons\/github.png",
                "id": 1
            }
        ],
        "ts": "1468586985.002316"
    },
    {
        "user": "U1S4E2S8L",
        "text": "<@U1S4E2S8L|karthikksiti> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468592127.002318"
    },
    {
        "user": "U1S565G4R",
        "text": "<@U1S565G4R|albertbarron> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468593618.002319"
    },
    {
        "user": "U1S54EHL7",
        "text": "<@U1S54EHL7|steven.lebowitz> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468594254.002320"
    },
    {
        "type": "message",
        "user": "U16FB85U6",
        "text": "Hi, anyone know how to schedule events (running contracts) daily (or at certain intervals) on Fabric?",
        "ts": "1468595924.002321"
    },
    {
        "user": "U1S6C2G5B",
        "text": "<@U1S6C2G5B|gemoore> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468600557.002322"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "<@U1LES78TT>: I am 22 comits behind master",
        "ts": "1468603043.002323"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "I did a merge to my branch Wed bc needed to submit a pull request",
        "ts": "1468603064.002324"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "not sure, at least this build won\u2019t complain if I submit erroneous invokations",
        "ts": "1468603081.002325"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "could be chaincode related",
        "ts": "1468603089.002326"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "I also had issues submitting chaincode locally, so far, only using github :.",
        "ts": "1468603105.002327"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": ":confused:",
        "ts": "1468603107.002328"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "hi..",
        "ts": "1468607441.002329"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "I have a question on membersrvc",
        "ts": "1468607454.002330"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "the example shows the configuration is in a file. Is that the only way to do it. Can we have our own security rather than using default membersrvc provided by hyperledger",
        "ts": "1468607512.002331"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "you mean like a plugin?",
        "ts": "1468608139.002332"
    },
    {
        "type": "message",
        "user": "U113EGLBE",
        "text": "<@U1LES78TT> The erorr you got: \n\n\"&lt;error&gt;Failed to launch chaincode spec(Error starting container: The command '\/bin\/sh -c go install <http:\/\/github.com\/IBM-Blockchain\/learn-chaincode\/blob\/master\/finished|github.com\/IBM-Blockchain\/learn-chaincode\/blob\/master\/finished> &amp;amp;&amp;amp; cp src\/github.com\/hyperledger\/fabric\/peer\/core.yaml $GOPATH\/bin &amp;amp;&amp;amp; mv $GOPATH\/bin\/finished $GOPATH\/bin\/842b56c1a37dbe53282da0885c126ad4d2e52b885a1ac577e7abd1cb6b8e4bb15ddced4db7d5c9f05f1afbb7d993448cc3ee9cf5d46eadb87f1a0f5d899254ed' returned a non-zero code: 2)&lt;\/error&gt;\"\n\nMeans your chaincode has a compile error so it can't deploy it. Go to your chaincode file and do a go build file.go\nFix any of your compile errors\/warnings and then try deploying again.",
        "ts": "1468609360.002333"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "<@U1LS81ER3>:  Yes correct..like a plugin can we  olug in any other identity management",
        "ts": "1468610681.002334"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "plug in",
        "ts": "1468610687.002335"
    },
    {
        "user": "U1S8D1X6V",
        "text": "<@U1S8D1X6V|merve> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468610895.002336"
    },
    {
        "type": "message",
        "user": "U1LS81ER3",
        "text": "mmm, my understanding is that the fabric is modular, however, right now only the consensus piece has been worked on, not sure about thememberservc",
        "ts": "1468612563.002337"
    },
    {
        "type": "message",
        "user": "U0KMJF9AA",
        "text": "ok..thanks",
        "ts": "1468613879.002338"
    },
    {
        "user": "U1S9KRK7F",
        "text": "<@U1S9KRK7F|hull> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468620262.002339"
    },
    {
        "user": "U1S9V7HAL",
        "text": "<@U1S9V7HAL|moriyama> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468624834.002340"
    },
    {
        "user": "U1SB979DF",
        "text": "<@U1SB979DF|larrylu2000> has joined the channel",
        "type": "message",
        "subtype": "channel_join",
        "ts": "1468651561.002341"
    }
]